### PR TITLE
Adding recipes for py-qutip and py-qutip-qip

### PR DESCRIPTION
**Pull request by marcodelapierre**
These are 2 related Quantum Computing packages

### CODE REVIEW

**Comment by marcodelapierre** [comment]
@adamjstewart pinging you on this one, thank you :-)

File: var/spack/repos/builtin/packages/py-qutip-qip/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    # using github for now, because pypi tarball is missing the VERSION file
+    # pypi = "qutip-qip/qutip-qip-0.2.2.tar.gz"
+
+    version("0.2.3", sha256="a6a3a549cf6983e3ecef2cf07d00be83c146321fb588e250a49d020788a4e590")
+    version("0.2.2", sha256="4a9c79bb31c2fb2c72428764b2a5f6d8b1c667cebc8257cce1395c7e87d11217")
+
+    depends_on("py-setuptools@42:", type="build")
+    depends_on("py-packaging", type="build")
+
+    depends_on("py-numpy@1.12:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qutip-qip/package.py]:
```suggestion
    depends_on("py-numpy@1.16.6:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-qutip-qip/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on("py-scipy@1.0:", type=("build", "run"))
+
+    depends_on("py-qutip@4.5:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qutip-qip/package.py]:
```suggestion
    depends_on("py-qutip@4.6:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-qutip-qip/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = "https://github.com/qutip/qutip-qip"
+    url = "https://github.com/qutip/qutip-qip/archive/refs/tags/v0.2.2.tar.gz"
+    # using github for now, because pypi tarball is missing the VERSION file
+    # pypi = "qutip-qip/qutip-qip-0.2.2.tar.gz"
+
+    version("0.2.3", sha256="a6a3a549cf6983e3ecef2cf07d00be83c146321fb588e250a49d020788a4e590")
+    version("0.2.2", sha256="4a9c79bb31c2fb2c72428764b2a5f6d8b1c667cebc8257cce1395c7e87d11217")
+
+    depends_on("py-setuptools@42:", type="build")
+    depends_on("py-packaging", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qutip-qip/package.py]:
```suggestion
    depends_on("py-packaging", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-qutip/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://qutip.org/"
+    pypi = "qutip/qutip-4.7.0.tar.gz"
+
+    version("4.7.1", sha256="9a87178e68b145c2145b526caa943ccc8400a111325ced45bd17f9b893663af2")
+    version("4.7.0", sha256="a9dde64457991ef1c5a7d4186b5348a16a71480a610f1c0902e4d656ddc12e31")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-packaging", type="build")
+
+    depends_on("py-cython@0.29.20:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qutip/package.py]:
```suggestion
    depends_on("py-cython@0.29.20:", type="build")
```

File: var/spack/repos/builtin/packages/py-qutip/package.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    """QuTiP: The Quantum Toolbox in Python"""
+
+    homepage = "https://qutip.org/"
+    pypi = "qutip/qutip-4.7.0.tar.gz"
+
+    version("4.7.1", sha256="9a87178e68b145c2145b526caa943ccc8400a111325ced45bd17f9b893663af2")
+    version("4.7.0", sha256="a9dde64457991ef1c5a7d4186b5348a16a71480a610f1c0902e4d656ddc12e31")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-packaging", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qutip/package.py]:
```suggestion
    depends_on("py-packaging", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by marcodelapierre** [comment]
I have applied the requested edits

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by adamjstewart — APPROVED** [review]

**Comment by marcodelapierre** [comment]
@spackbot run pipeline

**Pull request merged**